### DISCUSSION TITLE
예제 5-2 HTTP 응답 상태 코드 체크 ApiService.js 수정사항 (스프링 부트 2.7.0)

### DISCUSSION THREAD

**Discussion by conda-12** [fsoftwareengineer/todo-application#117] [질문과 답변]
```
return fetch(options.url, options)
  .then(response => {
  
      if (!response.ok) {
  
          return Promise.reject(response);
      }
  
      return response.json();
      }
  )
.catch(error => {

    if (error.status === 403) {
      
      window.location.href = "/login";
    }
    
    return Promise.reject(error);
    });
}

export function signin(userDTO) {
return call("/auth/signin", "POST", userDTO)
.then(response => {
    if (response.token) {
    
            localStorage.setItem(ACCESS_TOKEN, response.token);
    
            window.location.href = "/";
        }
    })
}
```

자료를 찾아보니 fetct API는 Response Object를 리턴 한다고 되어있습니다.
책에 나와있는 것처럼 만약 response Object를 .json() 메소드로 상태를 변경한 후 Promise.reject(json)울 넘긴다면
error.status 이 부분에서는 response Object가 가지고 있는 property인 response.status 접근이 불가능 할 것입니다.

참고 자료:
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
https://developer.mozilla.org/en-US/docs/Web/API/Response
https://developer.mozilla.org/en-US/docs/Web/API/Response/json

**Comment by KoreanJANG** [comment]
해당 내용과 관련이 있어 문의사항을 남깁니다~! 

현재 백엔드 프론트엔드를 aws에 다 올렸는데 동작에 이상이 있습니다. 
먼저 기본 URL을 입력하면 "로딩중" 페이지에서 순간 로그인 페이지로 이동해야 하고, 또한 /signup 등을 치면 
해당 페이지에는 접속이 되는데 버튼 이동 등은 되지 않습니다. 

따라서 f12로 에러 사항을 보니 각 페이지 마다 404에러고 ApiService.js 25번째 줄 쯤에 있는 fetch 가 문제라고 
되어 있습니다. 

해당 에러는 로컬에서는 발생하지 않았었고,  aws에 배포를 하니 나타나는 문제입니다. 

혹시 위에 분이 남겨주신 내용 때문에 이상 동작하는 것인지 문의드립니다!!  

가능하시면 수정해야 할 코드도 남겨주시면 감사드리겠습니다!@
